### PR TITLE
Rename `repo.GetOwner` to `repo.LoadOwner`

### LINKED ISSUE #22963: repo.GetOwner should be repo.LoadOwner
### Feature Description

I think the `GetOwner` here should be `LoadOwner`. `Get` means return something and `Load` means something is `nil`, we need to load it.
This func dosen't return anything, but the description of this func says `GetOwner returns the repository owner `
https://github.com/go-gitea/gitea/blob/8bba7e30b2833d750af0345b75a3998785ef55ef/models/repo/repo.go#L376-L384

In other places, the functions which do samething are called `LoadOwner`
https://github.com/go-gitea/gitea/blob/8bba7e30b2833d750af0345b75a3998785ef55ef/models/project/project.go#L110-L116
https://github.com/go-gitea/gitea/blob/8bba7e30b2833d750af0345b75a3998785ef55ef/models/admin/task.go#L95-L112

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by yp05327**
Fixes https://github.com/go-gitea/gitea/issues/22963

### CODE REVIEW

File: modules/repository/create.go [hunk 1]
```diff
@@ -379,7 +379,7 @@ func UpdateRepository(ctx context.Context, repo *repo_model.Repository, visibili
 	}
 
 	if visibilityChanged {
-		if err = repo.GetOwner(ctx); err != nil {
+		if err = repo.LoadOwner(ctx); err != nil {
 			return fmt.Errorf("getOwner: %w", err)
```
Comment (yardenshoham) [inline modules/repository/create.go]:
```suggestion
			return fmt.Errorf("LoadOwner: %w", err)
```

**Review by yardenshoham — APPROVED** [review]

File: models/git/protected_branch.go [hunk 2]
```diff
@@ -314,8 +314,8 @@ type WhitelistOptions struct {
 // This function also performs check if whitelist user and team's IDs have been changed
 // to avoid unnecessary whitelist delete and regenerate.
 func UpdateProtectBranch(ctx context.Context, repo *repo_model.Repository, protectBranch *ProtectedBranch, opts WhitelistOptions) (err error) {
-	if err = repo.GetOwner(ctx); err != nil {
-		return fmt.Errorf("GetOwner: %v", err)
+	if err = repo.LoadOwner(ctx); err != nil {
+		return fmt.Errorf("LoadOwner: %v", err)
```
Comment (delvh) [inline models/git/protected_branch.go:318]: How did that make it in?
I'll accept it for this PR as it seems to be a problem throughout this whole file…
It should be
```go
		return fmt.Errorf("LoadOwner: %w", err)
```

**Review by delvh — APPROVED** [review]

File: models/git/protected_branch.go (hunk 2 above)
Comment (yp05327) [reply models/git/protected_branch.go:318]: image
it seems that this problem is not only in this file.

Comment (delvh) [reply models/git/protected_branch.go:318]: I know.
(Although many of these will be `log.X("%v", err)`, where it won't apply.)
That's what I meant with "I accept it for this PR".

**Comment by lunny** [comment]
LG-TM bot

**Pull request merged**